### PR TITLE
 Blob trigger and bindings lesson (TypeScript)

### PR DESCRIPTION
**Pull request by lechnerc77**
According to issue #17 this pull request contains the lesson for Blob trigger and bindings in TypeScript.

It contains:
- The lesson
- The sample code
- The homework including solution
- VS Code Tour files
- Definition of VS Code Workspace

Due to the different options the structure deviates from the .NET session. I tried to keep it as close as possible.

### CODE REVIEW

**Comment by marcduiker** [comment]
Excellent, will review in a couple of days!

**Review by marcduiker — CHANGES_REQUESTED** [review]
Setting up the builds for the .NET lessons it also got me thinking about the folder structure for the typescript projects. For .NET I have one Function App project per lesson, that contains multiple functions. For TypeScript there are subfolders within the AzureFunctions.Blob folder that contains a new Function App project for each of the exercises. On one hand this is very explicit (and thus clear) but on the other hand there is a lot of duplication regarding the Function App level files now (hosts.json, package.json etc). And if we want to add project builds as PR check (as for .NET), it means we have to configure many builds per lesson which is not great maintenance wise.

What about we have just one Function App project for each lesson and that contains the subfolders for the various functions that maps to the exercises. This would require to name the functions differently though. I would be OK with naming the functions according to their exercise number such as `Exercise2PlainBlobOutputBinding` (I will then do the same thing for the .NET function names). What do think?

File: src/typescript/AzureFunctions.Blob/02UsingPlainBlobOutputBindings/.funcignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+*.js.map
+*.ts
+.git*
+.vscode
+local.settings.json
```
Comment (marcduiker) [inline src/typescript/AzureFunctions.Blob/02UsingPlainBlobOutputBindings/.funcignore:5]: I would be OK with including local.settings.json in source control as long as these files do not contain secrets. I see that you named the files `sample.local.settings.json` but I didn't see any instructions in the lesson to rename them. Ideally I'd prefer the project to build without any instructions for the user, so as long as the settings don't contain anything secret you can include them.

Comment (lechnerc77) [reply src/typescript/AzureFunctions.Blob/02UsingPlainBlobOutputBindings/.funcignore]: Done

File: lessons/typescript/blob/blob-lesson-ts.md [hunk 2]
```diff
@@ -1,3 +1,569 @@
...
+- A _blob_ in a container. A container can contain multiple blobs. You can also use _folders_ to structure the blobs in a container.
+
+We will mostly use a _local storage emulator_ instead of creating a storage account in Azure to deal with blobs. This is great for local development. In this section we will setup a sample container containing a folder and a blob.
+
+### Steps
+
+1. Install [Azurite](https://github.com/Azure/Azurite) as cross-platform emulator for the storage.
+2. Install [Azure Storage Explorer](https://azure.microsoft.com/features/storage-explorer/).
+3. Create a directory for the data storage of Azurite e.g. `C:\Users\<Your_Name>\azurite-storage`
+4. Start the Azure Storage Emulator with the command:
```
[the remark this answers is not in the document]
Comment (lechnerc77) [reply lessons/typescript/blob/blob-lesson-ts.md]: Done

**Comment by lechnerc77** [comment]
> Setting up the builds for the .NET lessons it also got me thinking about the folder structure for the typescript projects. For .NET I have one Function App project per lesson, that contains multiple functions. For TypeScript there are subfolders within the AzureFunctions.Blob folder that contains a new Function App project for each of the exercises. On one hand this is very explicit (and thus clear) but on the other hand there is a lot of duplication regarding the Function App level files now (hosts.json, package.json etc). And if we want to add project builds as PR check (as for .NET), it means we have to configure many builds per lesson which is not great maintenance wise.
> 
> What about we have just one Function App project for each lesson and that contains the subfolders for the various functions that maps to the exercises. This would require to name the functions differently though. I would be OK with naming the functions according to their exercise number such as `Exercise2PlainBlobOutputBinding` (I will then do the same thing for the .NET function names). What do think?

There a pros and cons for both ways. I think maintenance from your side should be as easy as possible, so I rearranged the files as described. This includes:

- source code per se (including additional comments in index.ts files)
- code tour files
- test files

I also had to change a dependency for the blob storage SDK as version 12.6.0 contains a bug - pinned it to 12.5.0 and adjusted the readme text

**Review by marcduiker — APPROVED** [review]
LGTM

**Pull request merged**